packages/fiori/src/ShellBar.ts

```diff
@@ -681,7 +744,7 @@ class ShellBar extends UI5Element {
 	_handleSizeS() {
 		const hasIcons = this.showNotifications || this.showProductSwitch || this.searchField.length || this.items.length;
 
-		const newItems = this._getAllItems(hasIcons).map(info => {
+		const newItems = this._getAllItems(!!hasIcons).map(info => {
```

Comment:
better move casting to boolean where the variables is declared (this.items.length => !!this.items.length)